Document existingJudgments parameter and judgment retry endpoint (#12862)
opensearch-project/documentation-website: _search-plugins/search-relevance/judgments.md

- For each query, the top k documents are retrieved using the defined search configuration, which includes the index information. The query and each document from the result list create a query-document pair.
- The LLM is then called with a predefined prompt to generate a judgment for each query-document pair.

> - All generated judgments are stored in the judgment list. To reuse ratings from previous judgments in a new run, use the `existingJudgments` parameter. For more information, see [Reusing existing judgments](#reusing-existing-judgments).

Reviewer: Let's remove this.
```suggestion
```

Author: Including in the new commit.

```
{% include copy-curl.html %}

#### Example response

```json
{
  "judgment_id": "b54f791a-3b02-49cb-a06c-46ab650b2ade",
  "status": "RETRYING",
  "message": "Retrying failed documents"
}
```

The retry runs asynchronously and generates new ratings for only the previously failed documents; ratings that already succeeded are left unchanged. The response returns immediately with a `status` of `RETRYING`. To track progress, retrieve the judgment list and check its `status`, as described in [Viewing a judgment list](#viewing-a-judgment-list). If individual documents still fail to get rated (for example, because the LLM provider is still throttling requests), the judgment list's `status` returns to `COMPLETED`---check the `failures` array and `metadata` field to see which documents are still unrated. If the retry process itself fails (for example, due to an internal error), the judgment list's `status` becomes `ERROR`.

#### Status values

The following table lists the possible `status` values for a judgment list.

| Status | Description |
| :--- | :--- |
| `PROCESSING` | The judgment list is being generated. |

> | `COMPLETED` | Generation (or a retry) has finished. Some documents may still be unrated---check the `failures` array and `metadata` field. |

Reviewer 2: ```suggestion
| `COMPLETED` | Generation (or a retry) has finished. Some documents may still be unrated: check the `failures` array and `metadata` field to verify the document ratings. |
```

| `ignoreFailure` | Boolean | Whether to continue processing other documents if the LLM fails to generate a judgment for some documents. Default is `false`. |
| `llmJudgmentRatingType` | String | The type of rating scale to use. Valid values are `SCORE0_1` (numeric scale 0--1) and `RELEVANT_IRRELEVANT` (binary relevant/irrelevant). Use `SCORE0_1` for graded relevance metrics such as NDCG. Use `RELEVANT_IRRELEVANT` for binary metrics such as precision and recall. |
| `promptTemplate` | String | Optional. A custom prompt template for the LLM. Supports {% raw %}`{{queryText}}`{% endraw %} and {% raw %}`{{hits}}`{% endraw %} placeholders. If not provided, the default template is used. |

> | `overwriteCache` | Boolean | Optional, deprecated. This parameter no longer has any effect because the global judgment cache has been removed. It is still accepted so that older clients don't break. To retry only the documents that failed in a completed judgment, use the [retry endpoint](#retrying-failed-documents). |

Reviewer: ```suggestion
| `existingJudgments` | Array of strings | Array of strings | Optional. A list of up to 5 existing judgment IDs whose ratings are reused. For each query-document pair, SRW checks these judgments in order, using the rating from the first match found. Only pairs without an existing rating in any of these judgments are sent to the LLM for evaluation. |
| `overwriteCache` | Boolean | Optional. Deprecated. Accepted but ignored. The global judgment cache has been removed. |
```

Reviewer: @kolchfa-aws would you make this change. Other than that it is ready for doc review.

| `ignoreFailure` | Boolean | Whether to continue processing other documents if the LLM fails to generate a judgment for some documents. Default is `false`. |
| `llmJudgmentRatingType` | String | The type of rating scale to use. Valid values are `SCORE0_1` (numeric scale 0--1) and `RELEVANT_IRRELEVANT` (binary relevant/irrelevant). Use `SCORE0_1` for graded relevance metrics such as NDCG. Use `RELEVANT_IRRELEVANT` for binary metrics such as precision and recall. |
| `promptTemplate` | String | Optional. A custom prompt template for the LLM. Supports {% raw %}`{{queryText}}`{% endraw %} and {% raw %}`{{hits}}`{% endraw %} placeholders. If not provided, the default template is used. |
| `existingJudgments` | Array of strings | Array of strings | Optional. A list of up to 5 existing judgment IDs whose ratings are reused. For each query-document pair, SRW checks these judgments in order, using the rating from the first match found. Only pairs without an existing rating in any of these judgments are sent to the LLM for evaluation. |

> | `overwriteCache` | Boolean | Optional. Deprecated. Accepted but ignored. The global judgment cache has been removed. |

Reviewer 2: Should we remove the overwriteCache parameter from the example in the LLM-as-a-judge tutorial? https://github.com/opensearch-project/documentation-website/blob/main/_tutorials/llm-as-a-judge-tutorial.md?plain=1#L183

```
{% include copy-curl.html %}

> ### Reusing existing judgments

Reviewer: Let's remove this entire part.

</details>

### Retrying failed documents

> A judgment can finish with a `status` of `COMPLETED` even if some documents didn't receive a rating, for example because the LLM provider throttled or timed out on those requests. Rather than regenerating the entire judgment, you can retry only the documents that failed by sending a `POST` request to the `_retry` endpoint with the judgment's ID.

Reviewer 2: Could you tell the user how to find out which documents didn't receive a rating? The "Viewing a judgment list" section doesn't show it either.

Reviewer 2: Because this section sits under Managing judgment lists next to view/delete/search, which all work for every judgment type, the placement reads as though retry does too. Either state the restriction in the opening sentence or move the section under Using LLM-as-a-Judge.

Reviewer 2: In terms of terminology, this section uses "judgment" to mean "judgment list", but we define judgment as a single rating for a single document. We should use "judgment list" in this section as well.

Author: Makes sense. 
Will update the above and create a new commit.

#### Endpoint

```json
POST _plugins/_search_relevance/judgments/{judgment_list_id}/_retry
```

#### Path parameters

The following table lists the available path parameters.

| Parameter | Data type | Description |
| :--- | :--- | :--- |
| `judgment_list_id` | String | The ID of the judgment list whose failed documents you want to retry. |

#### Example request

```json
POST _plugins/_search_relevance/judgments/b54f791a-3b02-49cb-a06c-46ab650b2ade/_retry
```
{% include copy-curl.html %}

#### Example response

```json
{
  "judgment_id": "b54f791a-3b02-49cb-a06c-46ab650b2ade",

>   "status": "RETRYING",

Reviewer 2: Can we add a table listing all possible status values and their descriptions?

  "message": "Retrying failed documents"
}
```

> The retry runs asynchronously and generates new ratings for only the previously failed documents; ratings that already succeeded are left unchanged. The response returns immediately with a `status` of `RETRYING`. To track progress, retrieve the judgment and check its `status`, as described in [Viewing a judgment list](#viewing-a-judgment-list). When the retry finishes, the judgment returns to a `status` of `COMPLETED`.

Reviewer 2: What if the retry fails? Does it still reach a `COMPLETED` status or does it stay in `ERROR`? So does `COMPLETED` mean that the retry finished but not necessarily that it succeeded so the users still need to check for failures?

Author: Yes, right. The status will be COMPLETED, but in the table, 'ERROR' will still be displayed.

#### Endpoint

```json
POST _plugins/_search_relevance/judgments/{judgment_list_id}/_retry
```

#### Path parameters

The following table lists the available path parameters.

| Parameter | Data type | Description |
| :--- | :--- | :--- |
| `judgment_list_id` | String | The ID of the judgment list whose failed documents you want to retry. |

#### Example request

```json
POST _plugins/_search_relevance/judgments/b54f791a-3b02-49cb-a06c-46ab650b2ade/_retry
```
{% include copy-curl.html %}

#### Example response

```json
{
  "judgment_id": "b54f791a-3b02-49cb-a06c-46ab650b2ade",
  "status": "RETRYING",
  "message": "Retrying failed documents"
}
```

The retry runs asynchronously and generates new ratings for only the previously failed documents; ratings that already succeeded are left unchanged. The response returns immediately with a `status` of `RETRYING`. To track progress, retrieve the judgment and check its `status`, as described in [Viewing a judgment list](#viewing-a-judgment-list). When the retry finishes, the judgment returns to a `status` of `COMPLETED`.

> In OpenSearch Dashboards, you can start a retry from the judgment list, which also shows whether each document is **Rated** or **Failed**.

Reviewer: Let's remove it.
```suggestion

```

Author: Including in the new commit.
